app/src/features/search/SearchPage.tsx

```diff
@@ -1,8 +1,171 @@
+   */
+  const collectFilterParams = useCallback((): IAdvancedSearch => {
+    if (location.search) {
+      const urlParams = qs.parse(location.search.replace('?', ''));
+
+      return { keywords: urlParams.keywords } as IAdvancedSearch;
+    }
+    return advancedSearchInitialValues;
+  }, [location.search]);
+
+  const [formikValues, setFormikValues] = useState<IAdvancedSearch>(collectFilterParams);
+  const formikRef = useRef<FormikProps<IAdvancedSearch>>(null);
+
+  /**
+   * Determines if the search parameters are empty
+   */
+  const isDefaultState = (): boolean => {
+    return (
+      JSON.stringify(!formikRef?.current || formikRef.current.values) === JSON.stringify(advancedSearchInitialValues)
+    );
+  };
+
+  /**
+   * Updates URL search params to reflect formikRef values
+   */
+  const updateSearchParams = () => {
+    const urlParams = qs.stringify(formikRef.current?.values);
+    history.push({
+      search: `?${urlParams}`
+    });
+  };
+
+  const handleResetSearchParams = () => {
+    history.push({
+      search: ``
+    });
+  };
+
+  const handleReset = async () => {
+    setFormikValues(advancedSearchInitialValues);
+    handleResetSearchParams();
+  };
+
+  const handleSubmit = async () => {
+    if (!formikRef?.current) {
+      return;
+    }
+
+    // empty Filters
+    if (isDefaultState()) {
+      return;
+    }
+
+    try {
+      searchDataLoader.refresh(formikRef.current?.values.keywords);
+
+      updateSearchParams();
+    } catch (error) {
+      const apiError = error as APIError;
+      showFilterErrorDialog({
+        dialogTitle: 'Error Searching Datasets',
+        dialogError: apiError?.message,
+        dialogErrorDetails: apiError?.errors
+      });
+    }
+  };
+
+  const showFilterErrorDialog = (textDialogProps?: Partial<IErrorDialogProps>) => {
+    dialogContext.setErrorDialog({
+      onClose: () => {
+        dialogContext.setErrorDialog({ open: false });
+      },
+      onOk: () => {
+        dialogContext.setErrorDialog({ open: false });
+      },
+      ...textDialogProps,
+      open: true
+    });
+  };
+
+  const results = (searchDataLoader.data || []).reduce((acc: any, item) => {
+    return [...acc, ...item.source.project];
+  }, []);
+
+  searchDataLoader.load(formikRef.current?.values.keywords || formikValues.keywords);
+
+  return (
+    <Box my={4}>
+      <Container maxWidth="xl">
+        <Box mb={5}>
+          <Box mb={1}>
+            <Typography variant="h1">Search</Typography>
+          </Box>
+          <Typography variant="body1" color="textSecondary">
+            BioHubBC Platform search.
+          </Typography>
+        </Box>
+        <Box>
+          <Formik<IAdvancedSearch>
+            innerRef={formikRef}
+            initialValues={formikValues}
+            onSubmit={handleSubmit}
+            onReset={handleReset}
+            enableReinitialize={true}>
+            <SearchComponent />
+          </Formik>
+        </Box>
+        <Box my={4}>
```

Comment:
Is it worth separating the result to another file for code cleanliness?  That way the result box operates as a dummy component that only takes in values to display.

Reply:
I think given the rudamentary nature of the results rendering logic, I think it's safe to keep it on the page for now.